### DISCUSSION TITLE
[Sybase] DDL change transaction mode to manual

### DISCUSSION THREAD

**Discussion by ehmatos** [dbeaver/dbeaver#13054] [Q&A] [answered]
When execute a DDL command, the transaction mode change to manual.
The opened transaction block de environment.

When the smart commit mode is enabled, the transaction mode shold be locked to auto-commit?

#### System information: 
- Operating system (distribution) and version: windows 10
- DBeaver version: Version 21.0.2.202104042040
- Additional extensions

#### Connection specification:
- Database name and version: Adaptive Server Enterprise/16.0 SP03 PL05/EBF 28234 SMP/P/sparc/SunOS 5.10/ase160sp03pl05x/3455/64-bit/FBO/Sat Jul 28 04:28:01 2018
- Driver name
- Do you use tunnels or proxies (SSH, SOCKS, etc)?

**Comment by uslss** [comment]
Seems it is expected behavior. You can learn more about smart commit mode [here](https://github.com/dbeaver/dbeaver/wiki/Auto-and-Manual-Commit-Modes#smart-commit-mode)

**Comment by ehmatos** [comment]
How to change this expected behavior? Locks in a production environment can be generated as a result of an improperly opened transaction. For example, when creating a local or global temporary table, a transaction was opened and it was just a question of getting data, it should never have been opened a transaction... Who can I request this change from?

**Comment by uslss** [comment, marked as answer]
you should switch to auto-commit mode. It is expected behavior for the [smart-commit mode](https://github.com/dbeaver/dbeaver/wiki/Auto-and-Manual-Commit-Modes#smart-commit-mode)
